Frontend/src/Components/Battery/Battery.js

```diff
@@ -0,0 +1,143 @@
+import {Flex, Center, Box, Spacer, Text, useColorMode} from "@chakra-ui/react";
+import RangeRow from "../Shared/RangeRow";
+import RangeBar from "../Shared/RangeBar";
+import CONSTANTS from "../../data-constants.json";
+import getColor from "../Shared/colors";
+
+export default function Battery(props) {
+    const { colorMode } = useColorMode();
+    const headerBg = getColor("header", colorMode);
+    const borderColor = getColor("border", colorMode);
+
+    const fanEnableText = (props.data?.fan_enable[0] ?? -1.0) ? 'EN' : 'DIS';
+    const fanEnableColor = (props.data?.fan_enable[0] ?? -1.0) ? getColor("greenBg", colorMode) : getColor("redBg", colorMode);
+    const fanEnableTextColor = (props.data?.fan_enable[0] ?? -1.0) ? getColor("txtGreenBg", colorMode) : getColor("txtRedBg", colorMode);
```

Comment:
```suggestion
    const fanEnableText = ((props.data?.fan_enable[0] ?? -1.0) === -1.0) ? 'N/A' : (props.data.fan_enable[0] ? 'EN' : 'DIS');
    const fanEnableColor = (props.data?.fan_enable[0] ?? false) ? getColor("greenBg", colorMode) : getColor("redBg", colorMode);
    const fanEnableTextColor = (props.data?.fan_enable[0] ?? false) ? getColor("txtGreenBg", colorMode) : getColor("txtRedBg", colorMode);
```
- Sorry, I know it wasn't in the design, but I think it would be a good idea to make the text 'N/A' when there is no value for fan_enable.
- Since the value of fan_enable and the default value if it doesn't exist aren't being compared to anything, we'll need to use `false` instead of -1.0 in the bottom two lines.